src/Idris/REPL.idr

```diff
@@ -932,6 +937,200 @@ process NOP
     = pure Done
 process ShowVersion
     = pure $ VersionIs  version
+process (ImportPackage package) = do
+  defs <- get Ctxt
+  let searchDirs = defs.options.dirs.extra_dirs
+  let Just packageDir = find
+        (\d => isInfixOf package (fromMaybe d (fileName d)))
+        searchDirs
+    | _ => pure (REPLError (pretty "Package not found in the known search directories"))
+  let packageDirPath = parse packageDir
+  tree <- coreLift $ explore packageDirPath
+  fentries <- unfoldTree (setRoot tree (parse ""))
+  errs <- for fentries \entry => do
+    let entry' = dropExtension entry
+    let sp = forget $ split (== dirSeparator) entry'
+    let ns = concat $ intersperse "." sp
+    let ns' = mkNamespace ns
+    catch (do addImport (MkImport emptyFC False (nsAsModuleIdent ns') ns'); pure Nothing)
+          (\err => pure (Just err))
+  let errs' = catMaybes errs
+  res <- case errs' of
+    [] => pure (pretty "Done")
+    onePlus => pure $ vsep !(traverse display onePlus)
+  pure (Printed res)
+ where
+  unfoldTree : {root : _} -> Tree root -> Core (List String)
+  unfoldTree (MkTree files subtrees) = do
+    (map toFilePath files ++) . concat <$> flip Core.traverse subtrees \(_ ** treeIO) => do
+      tree <- coreLift treeIO
+      unfoldTree tree
```

Comment:
This could be refactored in terms of `depthFirst`.

Reply:
Done